src/js/utils/BlobUtils.js

```diff
@@ -23,9 +23,28 @@
       callback(blob);
     },
 
-    canvasToBlob : function (canvas, callback, type /*, ...args*/) {
+    canvasToBlob : function (sourceCanvas, callback, type /*, ...args*/) {
       type = type || 'image/png';
 
+      var exportScaling = pskl.UserSettings.get(pskl.UserSettings.EXPORT_SCALING);
+      var canvas;
+      if (exportScaling > 1) {
+        // Scale the input canvas to an offscreen canvas.
+        canvas = document.createElement('canvas');
+        canvas.width = sourceCanvas.width * exportScaling;
+        canvas.height = sourceCanvas.height * exportScaling;
+
+        var ctx = canvas.getContext('2d');
+        ctx.mozImageSmoothingEnabled = false;
+        ctx.webkitImageSmoothingEnabled = false;
+        ctx.msImageSmoothingEnabled = false;
+        ctx.imageSmoothingEnabled = false;
+
+        ctx.drawImage(sourceCanvas, 0, 0, sourceCanvas.width, sourceCanvas.height, 0, 0, canvas.width, canvas.height);
+      } else {
+        canvas = sourceCanvas;
+      }
+
```

Comment:
You can reuse ImageResizer.resize to resize the canvas. The resizing should also be moved to the caller (ie PngExportController). CanvasToBlob is a helper and can not assume it is used for exporting.